esi/lib/node.py

```diff
@@ -0,0 +1,141 @@
+
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import concurrent.futures
+
+from esi import utils
+
+
+def network_list(connection, filter_node=None, filter_network=None):
+
+    def get_baremetal_info(filter_node):
+        baremetal_ports = None
+        baremetal_nodes = None
+
+        if filter_node:
+            baremetal_nodes = [connection.baremetal.find_node(name_or_id=filter_node,
+                                                              ignore_missing=False)]
+            baremetal_ports = [connection.list_nics_for_machine(uuid=baremetal_nodes[0].id)]
+        else:
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                f1 = executor.submit(connection.list_nics)
+                f2 = executor.submit(connection.list_machines)
+                baremetal_ports = f1.result()
+                baremetal_nodes = f2.result()
+        return baremetal_ports, baremetal_nodes
+
+    def get_network_info(filter_network):
```

Comment:
Similar comments to above, except I'd just move it into a `networks.py` file and maybe call it `network_and_port_list`